### PR TITLE
Column to whitelist in client

### PR DESCRIPTION
**Pull request by AxelVivien25**

### CODE REVIEW

File: mlflow_elasticsearchstore/elasticsearch_store.py [hunk 1]
```diff
@@ -60,26 +59,37 @@ def _hit_to_mlflow_experiment(self, hit: Any) -> Experiment:
...
+        return RunInfo(run_uuid=hit._id, run_id=hit._id,
+                       experiment_id=str(hit._source.experiment_id),
+                       user_id=hit._source.user_id,
+                       status=hit._source.status,
+                       start_time=hit._source.start_time,
+                       end_time=hit._source.end_time if hasattr(hit._source, 'end_time') else None,
+                       lifecycle_stage=hit._source.lifecycle_stage if
+                       hasattr(hit["_source"], 'lifecycle_stage') else None,
+                       artifact_uri=hit._source.artifact_uri
+                       if hasattr(hit["_source"], 'artifact_uri') else None)
```
Comment (t-henri) [inline mlflow_elasticsearchstore/elasticsearch_store.py:77]: Why do you sometimes access _source as if it was a key in a dict, and some other times as if it was a member of an object ? 
Also you used to access fields directly in the hit object, does the _source thing come from the elasticsearch version change ?

File: mlflow_elasticsearchstore/elasticsearch_store.py [hunk 2]
```diff
@@ -323,8 +333,24 @@ def list_all_columns(self, experiment_id: str, run_view_type: str) -> 'Columns':
...
+        columns_to_whitelist_key_dict: dict = {"metrics": set(), "params": set(), "tags": set()}
+        for col in columns_to_whitelist:
+            word = col.split(".")
+            key = ".".join(word[1:])
+            if word[0] == "metrics":
+                columns_to_whitelist_key_dict["metrics"].add(key)
+            elif word[0] == "params":
+                columns_to_whitelist_key_dict["params"].add(key)
+            elif word[0] == "tags":
+                columns_to_whitelist_key_dict["tags"].add(key)
```
Comment (t-henri) [inline mlflow_elasticsearchstore/elasticsearch_store.py:348]: Can be replaced by:
`columns_to_whitelist_key_dict[word[0]].add(key)`

File: mlflow_elasticsearchstore/elasticsearch_store.py [hunk 3]
```diff
@@ -375,8 +401,7 @@ def _get_orderby_clauses(self, order_by_list: List[str], s: Search) -> Search:
                     sort_clauses.append({key: {'order': sort_order}})
         sort_clauses.append({"start_time": {'order': "desc"}})
```
Comment (t-henri) [inline mlflow_elasticsearchstore/elasticsearch_store.py:402]: Probably for another review, but why do you always order by start_time desc ? In the UI we can chose to order by start time asc too

Comment (AxelVivien25) [reply mlflow_elasticsearchstore/elasticsearch_store.py:402]: Capture d’écran 2020-09-18 à 17 59 32


It is the default order_by and the one use for tie breaking

Comment (AxelVivien25) [reply mlflow_elasticsearchstore/elasticsearch_store.py:402]: But if I want to order by asc start time it works

**Review by t-henri — APPROVED** [review]

File: mlflow_elasticsearchstore/elasticsearch_store.py (hunk 1 above)
Comment (AxelVivien25) [reply mlflow_elasticsearchstore/elasticsearch_store.py]: Euh I don't know, it is fixed

**Pull request merged**